lxd/device/device_utils_disk.go

```diff
@@ -59,40 +65,43 @@ func DiskMount(srcPath string, dstPath string, readonly bool, recursive bool, pr
 
 	// Detect the filesystem
 	fstype := "none"
```

Comment:
Ok, so we should be able to shrink that change quite a bit.

`fstype` here should be removed and replaced where it's used by `fsName`.
You should also ensure that everywhere we call `DiskMount`, we pass `none` to the fsName argument, unless we want a specific filesystem (`ceph`).

The `IsBlockdev` logic seems reasonable enough in here, I don't think we need to move all of that under a check for `ceph`.